tests/test_monitors.py

```diff
@@ -7,12 +7,22 @@
     FinishReasonMonitor,
     ItemCountMonitor,
     ErrorCountMonitor,
+    WarningCountMonitor,
```

Comment:
**suggestion (non-blocking)**: This test module is becoming too long and it is difficult to find specific tests for specific monitors here. Maybe we could reorganize these tests so we have one test module for each monitor we are testing, so we can find the tests we need more easily. This certainly can be made in a different PR.

My initial idea would be replace this `test_monitors.py` file with several files organized like:
```
/tests
  /contrib
    /scrapy
      /monitors
        /test_warning_count_monitor.py
        /test_successful_requests_monitor.py
        /test_total_requests_monitor.py
        (etc etc)
```